nixpkgs.nix

```diff
@@ -25,5 +25,5 @@ if builtins.getEnv "STATIC_HASKELL_NIX_CI_NIXPKGS_UNSTABLE_BUILD" == "1"
     if builtins.pathExists ./nixpkgs/pkgs
       then import ./nixpkgs {}
       # Pinned nixpkgs version; should be kept up-to-date with our submodule.
-      # This is nixos-23.05 as of 2023-06-08, with minimal patches currently having open nixpkgs PR (see commits for PR links).
-      else import (fetchTarball https://github.com/nh2/nixpkgs/archive/9e49f8f1f37bc906cda1adb33064c325d760819a.tar.gz) {}
+      # This is nixos-23.11 as of 2024-01-01, with minimal patches currently having open nixpkgs PR (see commits for PR links).
+      else import (fetchTarball https://github.com/nh2/nixpkgs/archive/ca1f69776b9dda07656a1051310e5e138aa83067.tar.gz) {}
```

Comment:
I have started with your [static-haskell-nix-nixos-23.05](https://github.com/nh2/nixpkgs/tree/static-haskell-nix-nixos-23.05) branch, which included the following patches:

* https://github.com/nh2/nixpkgs/commit/9e49f8f1f37b **Keep, not merged**
* https://github.com/nh2/nixpkgs/commit/40d2db0cee0e merged in https://github.com/NixOS/nixpkgs/pull/243161
* https://github.com/nh2/nixpkgs/commit/75378f1bf479 merged in https://github.com/NixOS/nixpkgs/pull/243161
* https://github.com/nh2/nixpkgs/commit/c7e42e02e837 merged in https://github.com/NixOS/nixpkgs/pull/236700
* https://github.com/nh2/nixpkgs/commit/065f8ab1aeb3 merged in https://github.com/NixOS/nixpkgs/pull/236700

I then picked the current `nixos-23.11` (https://github.com/nixos/nixpkgs/commit/32f63574c85fbc80e4ba1fbb932cde9619bad25e) and cherry-picked https://github.com/nh2/nixpkgs/commit/9e49f8f1f37b on top.

I have a [`static-haskell-nix-nixos-23.11`](https://github.com/jonathanlking/nixpkgs/tree/static-haskell-nix-nixos-23.11) branch on my fork, which you could just copy over to yours if you're happy with these changes — I find it a bit odd that GitHub already allows accessing the archive under your fork but 🤷‍♂️

Reply:
I have also included https://github.com/NixOS/nixpkgs/pull/278099 so that `p11-kit` (which is a dependency of many things, including ghc) builds on single-user install non-NixOS systems.